.github/workflows/main.yml

```diff
@@ -1,225 +1,187 @@
-      - name: Use the cached venv
-        uses: actions/cache@v2
-        with:
-          path: ${{ github.workspace }}/venv
-          key: venv-${{ github.ref }}-${{ hashFiles('requirements.txt') }}
-      - name: Check import sorting
-        run: |
-          source ${{ github.workspace }}/venv/bin/activate
-          make sort
+      - uses: extractions/setup-just@aa5d15c144db4585980a44ebfdd2cf337c4f14cb  # v1.4.0
+      - name: Check js linting
+        run: just check-js
 
   test-py:
-    needs: [format-py, lint-py, sort]
+    needs: [check]
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v2
       - uses: "actions/setup-python@v2"
         with:
-          python-version: "3.6"
-      - name: Use the cached venv
-        uses: actions/cache@v2
+          python-version: "3.8"
+          cache: pip
+          cache-dependency-path: requirements.*.txt
+      - uses: extractions/setup-just@aa5d15c144db4585980a44ebfdd2cf337c4f14cb  # v1.4.0
+
+      - name: Retrieve assets
+        uses: actions/download-artifact@v2
         with:
-          path: ${{ github.workspace }}/venv
-          key: venv-${{ github.ref }}-${{ hashFiles('requirements.txt') }}
-      - name: Run tests
-        run: |
-          source ${{ github.workspace }}/venv/bin/activate
-          make test-py
+          name: node-assets
+          path: static/dist/js
 
-  install-js:
+      - name: Run python tests
+        run: just test-py
+
+  test-js:
+    needs: [check-js]
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v2
-      - name: Cache node modules
-        uses: actions/cache@v2
-        env:
-          cache-name: cache-node-modules
+      - uses: extractions/setup-just@aa5d15c144db4585980a44ebfdd2cf337c4f14cb  # v1.4.0
+
+      - name: Retrieve assets
+        uses: actions/download-artifact@v2
         with:
-          path: ./node_modules
-          key: install-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            install-${{ env.cache-name }}-
-            install-
-      - name: Install dependencies
-        run: npm install
+          name: node-assets
+          path: static/dist/js
+
+      - name: Run js tests
+        run: just test-js
 
-  format-js:
-    needs: [install-js]
+  lint-dockerfile:
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v2
-      - name: Setup node
-        uses: actions/setup-node@v1
-        with:
-          node-version: '12'
-      - name: Cache node modules
-        uses: actions/cache@v2
-        env:
-          cache-name: cache-node-modules
+      - uses: hadolint/hadolint-action@v1.4.0
         with:
-          path: ./node_modules
-          key: install-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
-      - run: make format-js
+          dockerfile: docker/Dockerfile
 
-  lint-js:
-    needs: [install-js]
+  docker-test:
+    needs: [test-py, test-js]
```

Comment:
Do we need the Python/JS tests to run before this job?  Or could we parallelise a bit here by letting this run without them as dependencies?

Reply:
This is weird - that `docker-test` shouldn't be a thing anymore

Reply:
Yeah, it's not there https://github.com/opensafely-core/opencodelists/blob/dokku/.github/workflows/main.yml
There is just `test` (which does both the python and js tests) and then the `deploy` step depends on that.   Odd that the PR has shown this in the diff

Reply:
Ah, classic "reading through the commits" issue, sorry for the noise!